### PR TITLE
feat(Communities): Improvements for token holders list 

### LINKED ISSUE #11858: Display all required contact data in token holders list
The token holder list in `SortableTokenHoldersList.qml` can display two types of entries:
- Unknown token holder, who is not a community member. For such holders only wallet address is displayed.
- Community member token holder.
Screenshot 2023-08-09 at 18 13 25

For second case we should add a contact details such as online status, ens, ringSpecModel and etc.

### PR DESCRIPTION
**Pull request by MishkaRogachev**
Close #11858

### What does the PR do

- [x] Refactor TokenHolderListItem to a separate file
- [x] Added the necessary  contact details (is contact, trust status and etc)
- [x] Fixed UI to match [fimga](https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?type=design&node-id=29468-617598&mode=design&t=9sNWRPqZdBhvlIJv-0) designs 
- [ ] live update contact data (Opened issue for that here: https://github.com/status-im/status-desktop/issues/11983)

### Affected areas

Communities, Community tokens

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

Screenshot 2023-08-19 at 16 21 54

### CODE REVIEW

**Comment by MishkaRogachev** [comment]
@caybro @alexandraB99 what approach are we currently using to notify the contact's data updated? is it still a [connection](https://github.com/status-im/status-desktop/blob/fac76299099c20ee314248f67a82bc6c192ac431/ui/app/mainui/activitycenter/views/ActivityNotificationMessage.qml#L59) to the contacts model?

File: src/app/modules/main/communities/tokens/models/token_item.nim [hunk 1]
```diff
@@ -43,7 +43,8 @@ proc initTokenItem*(
   result.remoteDestructedAddresses = remoteDestructedAddresses
   result.tokenOwnersModel = newTokenOwnersModel()
   result.tokenOwnersModel.setItems(tokenOwners.map(proc(owner: CommunityCollectibleOwner): TokenOwnersItem =
-    result = initTokenOwnersItem(owner.contactId, owner.name, owner.imageSource, owner.collectibleOwner, remoteDestructedAddresses)
+    # TODO: provide number of messages here
```
Comment (MishkaRogachev) [inline src/app/modules/main/communities/tokens/models/token_item.nim:46]: https://github.com/status-im/status-desktop/issues/11860

File: src/app/modules/main/module.nim [hunk 2]
```diff
@@ -981,11 +981,14 @@ method onCommunityMuted*[T](
 method getVerificationRequestFrom*[T](self: Module[T], publicKey: string): VerificationRequest =
   self.controller.getVerificationRequestFrom(publicKey)
 
-method getContactDetailsAsJson*[T](self: Module[T], publicKey: string, getVerificationRequest: bool): string =
+method getContactDetailsAsJson*[T](self: Module[T], publicKey: string, getVerificationRequest: bool, getOnlineStatus: bool): string =
   let contact = self.controller.getContact(publicKey)
   var requestStatus = 0
   if getVerificationRequest:
     requestStatus = self.getVerificationRequestFrom(publicKey).status.int
+  var onlineStatus = 0
```
Comment (endulab) [inline src/app/modules/main/module.nim:989]:
```suggestion
  var onlineStatus = OnlineStatus.Unknown
  if getOnlineStatus:
     onlineStatus = toOnlineStatus(self.controller.getStatusForContactWithId(publicKey).statusType)
  ...
  "onlineStatus": onlineStatus.int
```

File: ui/imports/utils/Utils.qml [hunk 3]
```diff
@@ -422,13 +422,14 @@ QtObject {
             removed: false,
             trustStatus: Constants.trustStatus.unknown,
             verificationStatus: Constants.verificationStatus.unverified,
-            incomingVerificationStatus: Constants.verificationStatus.unverified
+            incomingVerificationStatus: Constants.verificationStatus.unverified,
+            onlineStatus: 0
```
Comment (endulab) [inline ui/imports/utils/Utils.qml:426]: Constants.onlineStatus.inactive

File: src/app/modules/main/communities/tokens/models/token_item.nim (hunk 1 above)
Comment (MishkaRogachev) [reply src/app/modules/main/communities/tokens/models/token_item.nim:46]: https://github.com/status-im/status-desktop/issues/11983

File: ui/imports/utils/Utils.qml (hunk 3 above)
Comment (MishkaRogachev) [reply ui/imports/utils/Utils.qml]: fixed

**Review by endulab — APPROVED** [review]

**Review by jrainville — APPROVED** [review]
LGTM

**Pull request merged**